### PR TITLE
feat: button refactor standard markup

### LINKED ISSUE #417: button: compact button height is not matching with Fiori3 standards
**Describe the bug**
fundamental-ngx/core compact button height is not as per Fiori3 standards.

> Is this issue related to a specific component?
       Button component

> What version of the Fundamental Styles are you using?
       0.3.0

> What front-end framework are you implementing in, e.g., Angular, React, etc.?
Angular

**Expected behavior**

- compact button height should be 26px including border.

**Screenshots**
In fundamental-ngx

Screen Shot 2019-11-07 at 2 21 09 PM

In Fiori3

Screen Shot 2019-11-07 at 2 21 42 PM

### LINKED ISSUE #602: fix(split button): With Menu option should have a fixed max-width as 12rem as per Fiori3
image

**Is your feature request related to a problem? Please describe.**
Split button with menu should have a fixed max-width as per Fiori3 standards.

> Is this issue related to a specific component?
split-button

### PR DESCRIPTION
**Pull request by rengare**
## Related Issue
Closes #417
Closes #602

## Description
- added properties and values for button, segmented-button, button-split to match Fiori 3
- documentation updated
- test scenarios simplified
- `button-group` renamed to `segmented-button`
- according to BEM methodology, button must start with `fd-button` class
- variable renamed from `$fd-animation--speed` to `$fd-animation-speed` in multiple files

## BREAKING CHANGES:
- Removed class: `fd-button--half`
- Renamed class: before `fd-button-group` after `fd-button-split`
- the button type classes are not standalone, the require the base class `fd-button`. Example:
before 
```
<button class="fd-button--emphasized"... >Click me</button>
```
After
```
<button class="fd-button fd-button--emphasized"... >Click me</button>
```

### Before:
image

### After:
image

### Before
image

### After
image

### Before
image

### After
image

### CODE REVIEW

File: src/button-split.scss [hunk 1]
```diff
@@ -7,54 +7,73 @@
     .fd-button
 */
 $block: #{$fd-namespace}-button-split;
+$fd-button-split-corner-radius: var(--sapButton_BorderCornerRadius);
+$fd-button-split-border-width: var(--sapButton_BorderWidth);
+$fd-button-split-margin: 0.125 / 2;
```
Comment (JKMarkowski) [inline src/button-split.scss]: How does it know that it's REM?

File: package.json [hunk 2]
```diff
@@ -29,7 +29,7 @@
     "test:ci-config": "./ci-scripts/generateBackstopConfig.js",
     "test:reference": "backstop reference --config ./test/visual-regression-tests/config/backstopConfig.js --docker",
     "test:update": "npm run test:linux-fix && server-test start:playground http-get://localhost:3030 test:reference && backstop approve --config ./test/visual-regression-tests/config/backstopConfig.js",
-    "test:update-linux": "npm run test:update && sudo chown -R ${USER}:${USER} .",
+    "test:update-linux": "npm run test:linux-fix && server-test start:playground http-get://localhost:3030 test:reference && sudo chown -R ${USER}:${USER} . && backstop approve --config ./test/visual-regression-tests/config/backstopConfig.js",
```
Comment (JKMarkowski) [inline package.json]: Is it intendent?

**Comment by droshev** [comment]
@rengare can you add a little description of the purpose of each button type?

File: package.json (hunk 2 above)
Comment (rengare) [reply package.json]: yes, previously it backstop approve fails because files were owned by root (docker changed that)

**Comment by bcullman** [comment]
can we get some before and after screen caps?

also, pick a name:

image

**Comment by mikerodonnell89** [comment]
Didn't look closely at the CSS but docs look good to me

File: src/button-split.scss (hunk 1 above)
Comment (rengare) [reply src/button-split.scss]: fixed

**Pull request merged**